ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PreExecutingSubmissionValidator.scala

```diff
@@ -52,7 +53,7 @@ class PreExecutingSubmissionValidator[StateValue, ReadSet, WriteSet](
       for {
         decodedSubmission <- decodeSubmission(submissionEnvelope)
         fetchedInputs <- fetchSubmissionInputs(decodedSubmission, ledgerStateReader)
-        inputState = fetchedInputs.mapValues(hasDamlStateValue.damlStateValue)
+        inputState = fetchedInputs.view.mapValues(hasDamlStateValue.damlStateValue).toMap
```

Comment:
This _may_ be an improvement. We tried to remove usages of `.mapValues` in the past because we found out it was recomputing it at every usage of the resulting collection. I believe the new Collections API should have fixed it.

Reply:
yeah the new API makes more sense to me.

Reply:
https://github.com/scala/scala/blob/v2.13.4/src/library/scala/collection/IterableOnce.scala#L1255 calls `Map.from` which forces the thing. iirc the deprecation warning mentions that they will add a strict version of `mapValues` in the future. Haven’t checked if it’s in Scala 3.